app/broadcast_message/rest.py

```diff
@@ -0,0 +1,123 @@
+    # we might need stuff like template name for showing on the dashboard.
+    # TODO: should this paginate or filter on dates or anything?
+    broadcast_messages = [o.serialize() for o in service.broadcast_messages]
+    return jsonify(broadcast_messages=broadcast_messages)
+
+
+@broadcast_message_blueprint.route('/<uuid:broadcast_message_id>', methods=['GET'])
+def get_broadcast_message(service_id, broadcast_message_id):
+    # TODO: should this return template content/data in some way?
+    return jsonify(dao_get_broadcast_message_by_id_and_service_id(broadcast_message_id, service_id).serialize())
+
+
+@broadcast_message_blueprint.route('', methods=['POST'])
+def create_broadcast_message(service_id):
+    data = request.get_json()
+
+    validate(data, create_broadcast_message_schema)
+
+    service = dao_fetch_service_by_id(data['service_id'])
+    user = get_user_by_id(data['created_by'])
+    template = dao_get_template_by_id_and_service_id(data['service_id'], data['template_id'])
+
+    broadcast_message = BroadcastMessage(
+        service_id=service.id,
+        template_id=template.id,
+        template_version=template.version,
+        personalisation=data['personalisation'],
+        status=BroadcastStatusType.DRAFT,
+        starts_at=_parse_nullable_datetime(data.get('starts_at')),
+        finishes_at=_parse_nullable_datetime(data.get('finishes_at')),
+        created_by_id=user.id,
+    )
+
+    dao_create_broadcast_message(broadcast_message)
+
+    return jsonify(data=broadcast_message.serialize()), 201
+
+
+@broadcast_message_blueprint.route('/<uuid:broadcast_message_id>', methods=['POST'])
+def update_broadcast_message(service_id, broadcast_message_id):
+    data = request.get_json()
+
+    validate(data, update_broadcast_message_schema)
+
+    broadcast_message = dao_get_broadcast_message_by_id_and_service_id(broadcast_message_id, service_id)
+
+    if 'personalisation' in data:
+        broadcast_message.personalisation = data['personalisation']
+    if 'starts_at' in data:
+        broadcast_message.starts_at = _parse_nullable_datetime(data['starts_at'])
+    if 'finishes_at' in data:
+        broadcast_message.starts_at = _parse_nullable_datetime(data['finishes_at'])
+
+    dao_update_broadcast_message(broadcast_message)
+
+    return jsonify(data=broadcast_message.serialize()), 200
+
+
+@broadcast_message_blueprint.route('/<uuid:broadcast_message_id>/status', methods=['POST'])
+def update_broadcast_message_status(service_id, broadcast_message_id):
+    data = request.get_json()
+
+    validate(data, update_broadcast_message_status_schema)
+    broadcast_message = dao_get_broadcast_message_by_id_and_service_id(broadcast_message_id, service_id)
+
+    new_status = data['status']
+
+    # TODO: Restrict status transitions
+    # TODO: Do we need to validate that the user belongs to the same service, isn't the creator, has permissions, etc?
+    # or is that admin's job
```

Comment:
I think we should have the safety of checking in both place. But fine to leave as a TODO for now.